File: Project4/src/main/java/com/example/Project4/payload/auth/RegisterRequest.java
package com.example.Project4.payload.auth;


import java.time.LocalDateTime;

import lombok.*;

@Getter
@Setter
public class RegisterRequest {
    private String username;
    private String firstname;
    private String lastname;
    private String email;
    private String password;
    private LocalDateTime dob;
    private int gender;
    private String phone;
    private String token;
    private String pinCode;
    private int bmmid;
}

File: projectflutter/lib/data/auth/request/register_request.dart
class RegisterRequest {
  final String username;
  final String firstname;
  final String lastname;
  final String email;
  final String password;
  final String address;
  final DateTime dob;
  final int gender;
  final String phone;

  RegisterRequest(
      {required this.username,
      required this.firstname,
      required this.lastname,
      required this.email,
      required this.password,
      required this.address,
      required this.dob,
      required this.gender,
      required this.phone});

  Map<String, dynamic> toJson() => {
    'username': username,
    'firstname': firstname,
    'lastname': lastname,
    'email': email,
    'password': password,
    'dob': dob.toIso8601String(),
    'address': address,
    'gender': gender,
    'phone': phone,
  };
}
